kubernetes/resource_kubernetes_daemon_set_v1_test.go

```diff
@@ -74,6 +74,7 @@ func TestAccKubernetesDaemonSetV1_basic(t *testing.T) {
 					resource.TestCheckResourceAttr(resourceName, "spec.0.template.0.spec.0.container.0.name", "tf-acc-test"),
 					resource.TestCheckResourceAttr(resourceName, "spec.0.strategy.0.type", "RollingUpdate"),
 					resource.TestCheckResourceAttr(resourceName, "spec.0.strategy.0.rolling_update.0.max_unavailable", "1"),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.strategy.0.rolling_update.0.max_surge", "0"),
```

Comment:
This one is fine since it validates the default value, however, you need to write a separate test to validate at least 2 scenarios: when `max_surge` is set as an absolute number and when it uses a percentage notation.